lab_1/main.py

```diff
@@ -86,10 +146,41 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict)\
+        and isinstance(top_n, int):
+        lang_adv = {}
+        cross = []
+        lang_adv['name'] = profile_to_compare['name']
+        for i in get_top_n_words(profile_to_compare['freq'], top_n):
+            if i in get_top_n_words(unknown_profile['freq'], top_n):
+                cross.append(i)
+            else:
+                continue
+        length = 0
+        for words in profile_to_compare['freq']:
+            length += len(words)
+        max_word = 0
+        min_word = 0
+        for max_length in profile_to_compare['freq']:
+            if len(max_length) == max(map(len, profile_to_compare['freq'])):
```

Comment:
can you explain what happens here? in writing

Reply:
Sorry, but there I'll write in russian to avoid mistakes and misunderstanding
Данный цикл идет по всем ключам словаря freq в profile_to_compare (так как именно в ключах записаны слова, в значениях же - количество их использования) и сравнивает длину данного слова с максимальной длиной слова по всему словарю (что задается функцией max(map(len, profile_to_compare['freq'])), где при помощи функции map находится длина каждого слова словаря (функция len)  и затем берется максимальное значение функцией max), если же значения совпадают, то в переменную max_word присваивается само слово, чья длина и оказалась максимальной. То же самое происходит и в следующем цикле для нахождения самого короткого слова (min_word).